code/game/objects/items/melee/baton.dm

```diff
@@ -534,12 +576,19 @@
 
 /obj/item/melee/baton/security/update_icon_state()
 	if(active)
-		icon_state = "[initial(icon_state)]_active"
+		icon_state = "[base_icon_state]_active"
+		if(active_changes_inhand)
+			if(tip_changes_color)
+				inhand_icon_state = "[base_icon_state]_active_[set_baton_tip_color()]"
```

Comment:
Would you mind adding the setlight back here so that the glow color properly reflects the tip's color? ❤️ 
(The previous review for having it changed tip color didn't say remove the glow, just that it wasn't noticable enough on it's own.) :>

Reply:
Easy enough.

Reply:
There we go, now it does both.